pkg/compactor/block_visit_marker.go

```diff
@@ -62,32 +64,35 @@ func ReadBlockVisitMarker(ctx context.Context, bkt objstore.Bucket, blockID stri
 	return &blockVisitMarker, nil
 }
 
-func UpdateBlockVisitMarker(ctx context.Context, bkt objstore.Bucket, blockID string, compactorID string, blockVisitMarkerWriteFailed prometheus.Counter) error {
+func UpdateBlockVisitMarker(ctx context.Context, bkt objstore.Bucket, blockID string, reader io.Reader, blockVisitMarkerWriteFailed prometheus.Counter) error {
 	blockVisitMarkerFilePath := path.Join(blockID, BlockVisitMarkerFile)
-	blockVisitMarker := BlockVisitMarker{
-		CompactorID: compactorID,
-		VisitTime:   time.Now(),
-	}
-	visitMarkerFileContent, err := json.Marshal(blockVisitMarker)
```

Comment:
Insead of marshalling the same content multiple times for different blocks, we just pass in a reader